system/scripts/zone/ais/basic_monster.cs

```diff
@@ -50,24 +52,29 @@ protected IEnumerable Idle()
 	protected IEnumerable Attack()
 	{
 		SetRunning(true);
-
 		while (!target.IsDead)
 		{
 			if (!TryGetRandomSkill(out var skill))
 			{
 				yield return Wait(2000);
 				continue;
 			}
-
-			while (!InRangeOf(target, skill.GetAttackRange()))
-				yield return MoveTo(target.Position, wait: false);
-
+			var attackRange = skill.GetAttackRange();
```

Comment:
I feel like this might be a bit much to put into the AI script, especially if it would potentially have to be copied to other AIs. Perhaps it would be better to move it to `AiScript`, as a routine to get into attacking position? Something for us to think about.

Reply:
You're right in saying that this might not become a one-size-fits-all method, because different types of mobs will need other behavior and will then get their own AIs, but don't forget that this will also not stay the only melee AI script for normal monsters.

Not only are there hundreds of official AIs we might eventually want/need to reproduce, it was also always the intention to facilitate the creation of more varied custom AIs, to give monsters more of an identity, so they don't all act the exact same. And anyone who will write a custom AI in the future, ourselves included, might appreciate a simple to use method, or a collection of them, to easily get their monster into a certain attacking position.

What do you think?

Reply:
Sounds good, i'll create a new method for it

Reply:
Oh, also, I think it would be nice if we kept the scripts as simple and readable as possible, as they're typically intended to be customizable by beginners. `MoveRandom` for example is only used once in the basic AI as well, and who knows if other AIs are going to use it as is, but having that raw code inside the script would make that process much less straightforward.

Reply:
I've added MoveToAttackPosition() method to simplify the script.